src/users/users.service.spec.ts

```diff
@@ -36,53 +42,234 @@ describe("UsersService Unit Tests", () => {
+                await usersService.create(consts.CREATE_USER_DTO_TEST_OBJ_WITH_INVALID_EMAIL_FIELD);
+            }
+            catch (e) {
+                expect(e.message).toBe("Validation error: Validation isEmail on email failed");
+            }
+        });
+
+        // Tests the create function with dto that has no password field
+        it("should test create function with missing password field", async () => {
+            expect.assertions(1);
+            try {
+                await usersService.create(consts.CREATE_USER_DTO_TEST_OBJ_WITH_MISSING_PASSWORD_FIELD);
+            }
+            catch (e) {
+                expect(e.message).toBe("data and salt arguments required");
+            }
+        });
 
-    // Tests the findByEmail function
-    it("should findByEmail", async () => {
-        const user = await usersService.findByEmail(consts.TEST_USER.email)
-        expect(user.email).toEqual(consts.USER_ONE_DTO.email);
+        // Tests the create function with dto that has no passwordConfirmation field
+        it("should test create function with missing password confirmation field", async () => {
+            expect.assertions(1);
+            try {
+                await usersService.create(consts.CREATE_USER_DTO_TEST_OBJ_WITH_MISSING_PASSWORD_CONFIRMATION_FIELD);
+            }
+            catch (e) {
+                expect(e.message).toBe("Validation error");
+            }
+        });
     });
 
-    // Tests the update function (Successful update)
-    it("should update everything", async () => {
-        const updatedUser = await usersService.update(userID, consts.UPDATE_USER_DTO_TEST_OBJ);
-        expect(updatedUser.email).toEqual(consts.UPDATE_USER_DTO_TEST_OBJ.email);
+    describe("FindAll function", () => {
+        // Tests the findAll function
+        it("should findAll", async () => {
+            const userdtoArray = await usersService.findAll();
+
+            expect(userdtoArray[userdtoArray.length-1].email).toEqual(consts.USER_DTO_ARRAY[0].email);
+            expect(userdtoArray[userdtoArray.length-1].firstName).toEqual(consts.USER_DTO_ARRAY[0].firstName);
+            expect(userdtoArray[userdtoArray.length-1].lastName).toEqual(consts.USER_DTO_ARRAY[0].lastName);
+            expect(userdtoArray[userdtoArray.length-1].title).toEqual(consts.USER_DTO_ARRAY[0].title);
+            expect(userdtoArray[userdtoArray.length-1].organization).toEqual(consts.USER_DTO_ARRAY[0].organization);
+
+            /* Expect that the dates equals each other for the createdAt value. Can't just see
+                if the createdAt equal eachother due to time created being off by a couple of seconds */
+            expect(userdtoArray[userdtoArray.length-1].createdAt.getDate()).toEqual(consts.USER_DTO_ARRAY[0].createdAt.getDate());
+
+            /* The id of the test constant is undefined, so if 
+                findAll is successful, the id's should not be equal */
+            let idFlag: boolean;
```

Comment:
I think that the tests that are modeled after this (checking to see if this flag is false with toBe can be simplified with https://jestjs.io/docs/en/expect#tobefalsy